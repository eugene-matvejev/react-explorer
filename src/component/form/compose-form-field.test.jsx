import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import compose from './compose-form-field';

configure({ adapter: new Adapter() });

const FormField = compose((props) => <input {...props} />);

describe('<FormField/>', () => {
    const props = {
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<FormField {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['errors', ['{{error}}']],
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<FormField {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });
});
