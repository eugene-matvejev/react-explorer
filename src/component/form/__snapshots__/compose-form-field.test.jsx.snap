// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField/> render with default/required props 1`] = `
<label
  className="form-input  form-input_label"
>
  <Component
    data-cy=""
  />
</label>
`;

exports[`<FormField/> render with optional props [::errors] as "{{error}}" 1`] = `
<label
  className="form-input  form-input_label"
>
  <Component
    data-cy=""
  />
  <ul
    className="form-input_errors"
  >
    <li
      className="form-input_error"
      data-cy="-error-0"
      key="0"
    >
      {{error}}
    </li>
  </ul>
</label>
`;
